### PR TITLE
Use unique name for lookup ops

### LINKED ISSUE #19528: Restore problem when work with multiple tf.contrib.lookup.MutableHashTable
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 3.10.0-693.2.2.el7.x86_64
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: Python 2.7.14 :: Anaconda
- **Bazel version (if compiling from source)**: None
- **GCC/Compiler version (if compiling from source)**: None
- **CUDA/cuDNN version**: None
- **GPU model and memory**: None
- **Exact command to reproduce**: See below

### Describe the problem
We try to use multiple tf.contrib.lookup.MutableHashTable objects in the script. But after do checkpoint with Saver.save(), we find that it can not restore table contents successfully. 

The problem disappears if we explicitly specify name field for tables. It seems that the MutableHashTable save contents with spec key "xxxx-keys" and "xxxx-values", where "xxxx" is the object's name field defaults to "MutableHashTable".

https://github.com/tensorflow/tensorflow/blob/93bc2e2072e0daccbcff7a90d397b704a9e8f778/tensorflow/contrib/lookup/lookup_ops.py#L463-L467

Maybe it is better to associate unique name for each table object if not explicitly specified?

### Source code / logs
Create and insert tables, and save
```python
keys = tf.placeholder(dtype=tf.string, shape=[None])
values = tf.placeholder(dtype=tf.int64, shape=[None])
table1 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
table2 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
insert_table1 = table1.insert(keys, values)
insert_table2 = table2.insert(keys, values)
saver = tf.train.Saver()
with tf.Session() as sess:
    sess.run(insert_table1, feed_dict={keys: ["a"], values: [1]})
    sess.run(insert_table2, feed_dict={keys: ["b"], values: [2]})
    print "table1:", sess.run(table1.export())
    print "table2:", sess.run(table2.export())
    saver.save(sess, "checkpoint/test")
```
table1: (array(['a'], dtype=object), array([1]))
table2: (array(['b'], dtype=object), array([2]))

Try restore contents
```python
table1 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
table2 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
saver = tf.train.Saver()
with tf.Session() as sess:
    saver.restore(sess, "checkpoint/test")
    print "table1:", sess.run(table1.export())
    print "table2:", sess.run(table2.export())
```
table1: (array([], dtype=object), array([], dtype=int64))
table2: (array(['b'], dtype=object), array([2]))

### PR DESCRIPTION
**Pull request by tobegit3hub**
Add the unique names for lookup ops so that we don't need to specify op names when saving and restoring checkpoints.

This PR resolves https://github.com/tensorflow/tensorflow/issues/19528 .

### CODE REVIEW

**Comment by tobegit3hub** [comment]
Can you help to review this? @ysuematsu @angersson

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @ysuematsu: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by qlzh727** [comment]
@andreasst, can you also take a look as well?

File: tensorflow/contrib/lookup/lookup_ops.py [hunk 1]
```diff
@@ -347,21 +348,21 @@ def __init__(self,
...
           value_shape=self._default_value.get_shape(),
-          name=name)
+          name=self._name)
     super(MutableHashTable, self).__init__(key_dtype, value_dtype,
                                            self._table_ref.op.name.split(
                                                "/")[-1])
 
     if checkpoint:
-      saveable = MutableHashTable._Saveable(self, name)
+      saveable = MutableHashTable._Saveable(self, self._name)
```
Comment (andreasst) [inline tensorflow/contrib/lookup/lookup_ops.py]: I believe this means that previous checkpoints cannot be read any more.

**Comment by tensorflowbutler** [comment]
It has been 20 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

File: tensorflow/contrib/lookup/lookup_ops.py (hunk 1 above)
Comment (tobegit3hub) [reply tensorflow/contrib/lookup/lookup_ops.py]: If we use the same graph to load the checkpoint, `self._name` should be the same.
Will this break the functionality of checkpoint?

**Comment by qlzh727** [comment]
@tobegit3hub, please also rebase and resolve the merge conflict here. Thanks.

**Comment by tobegit3hub** [comment]
Try to fix the conflict by rebasing master and sorry for bring all the reviewer 😞

**Review by alextp — CHANGES_REQUESTED** [review]
Can you write a test that fails before this change but passes after this change?

This will help prevent the desired behavior from being silently dropped by future changes to the codebase.

File: tensorflow/contrib/lookup/lookup_ops.py [hunk 2]
```diff
@@ -357,24 +358,26 @@ def __init__(self,
...
       self._table_ref = gen_lookup_ops.mutable_hash_table_of_tensors_v2(
           shared_name=shared_name,
           use_node_name_sharing=use_node_name_sharing,
           key_dtype=key_dtype,
           value_dtype=value_dtype,
           value_shape=self._default_value.get_shape(),
-          name=name)
+          name=self._name)
+
     if executing_eagerly:
```
Comment (allenlavoie) [inline tensorflow/contrib/lookup/lookup_ops.py]: Can we replace the op_name (this block) with your unique name now?

I'd suggest keeping it as a method-local variable until it gets passed to super().__init__, since it looks like that sets self._name (and it would be unfortunate to have multiple names).

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 28 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
